File: android/api-tests/src/test/java/com/revenuecat/apitests/java/MapperHelpersApiTests.java
package com.revenuecat.apitests.java.mappers;

import com.revenuecat.purchases.hybridcommon.mappers.MappersHelpersKt;

import org.json.JSONArray;
import org.json.JSONObject;

import java.util.List;
import java.util.Map;

@SuppressWarnings("unused")
class MapperHelpersApiTests {
    private void checkConvertToJson(Map<String, Object> map) {
        JSONObject json = MappersHelpersKt.convertToJson(map);
    }

    private void checkConvertToJsonArray(List<Object> list) {
        JSONArray jsonArray = MappersHelpersKt.convertToJsonArray(list);
    }

    private void checkConvertToMap(JSONObject json) {
        Map<String, String> map = MappersHelpersKt.convertToMap(json);
    }
}

File: android/api-tests/src/test/java/com/revenuecat/apitests/kotlin/mappers/MapperHelpersApiTests.kt
package com.revenuecat.apitests.kotlin.mappers

import com.revenuecat.purchases.hybridcommon.mappers.convertToJson
import com.revenuecat.purchases.hybridcommon.mappers.convertToJsonArray
import com.revenuecat.purchases.hybridcommon.mappers.convertToMap
import org.json.JSONArray
import org.json.JSONObject

@Suppress("unused", "UNUSED_VARIABLE")
private class MapperHelpersApiTests {
    fun checkConvertToJson(map: Map<String, *>) {
        val json: JSONObject = map.convertToJson()
    }

    fun checkConvertToJsonArray(list: List<*>) {
        val jsonArray: JSONArray = list.convertToJsonArray()
    }

    fun checkConvertToMap(json: JSONObject) {
        val map: Map<String, String?> = json.convertToMap()
    }
}
